src/org/labkey/test/pages/ImportDataPage.java

```diff
@@ -173,13 +253,21 @@ public ElementCache()
         WebElement uploadFileDiv = Locator.byClass("panel-body").findWhenNeeded(uploadPanel);
         WebElement uploadExpando = Locator.byClass("lk-import-expando").findWhenNeeded(uploadPanel);
         WebElement uploadFileFilePath = Locator.input("file").findWhenNeeded(uploadFileDiv);
+        RadioButton addRowsFile = RadioButton.RadioButton().withLabelContaining("Add").findWhenNeeded(uploadPanel);
+        RadioButton updateRowsFile = RadioButton.RadioButton().withLabelContaining("Update").findWhenNeeded(uploadPanel);
+        Checkbox mergeFile = Ext4Checkbox().withLabelContaining("Allow new").findWhenNeeded(uploadPanel);
+        Checkbox mergeFileOptional = Ext4Checkbox().withLabelContaining("Allow new").findOrNull(uploadPanel);
```

Comment:
It is not a big deal, and entirely up to you if you want to change it, but if you call findWhenNeeded or refindWhenNeeded you can check for visibility of the component which would then change isFileMergeOptionPresent to:

```
    public boolean isFileMergeOptionPresent()
    {
        return elementCache().mergeFileOptional.isVisible();
    }
```
One thing using findWhenNeeded does help with is if in the future the the code is changed to hide or show the control through css attributes this will still work. Using findOfNull would find a control this is present but not visible.

Reply:
@labkey-danield I think you meant `elementCache().mergeFile.getComponentElement().isVisible()`.
The `Checkbox` component doesn't have an `isVisible` method.

Reply:
I keep forgetting that.
In that case I wouldn't change it from the findOrNull implementation.